File: implementations/react/src/patterns/behavioral/mediator/concrete-colleague-b.ts
import { Colleague } from "./colleague";
import type { Item } from "./types/type";

export class ResultsList extends Colleague {
  view: Item[] = [];
  operationB(show: Item[]) {
    this.view = show;
    this.m.notify(this, "B:RESULTS_RENDERED", show.length);
  }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/mediator/concrete_colleague_b.dart
import 'colleague.dart';
import 'types.dart';

class ResultsList extends Colleague {
  List<Item> view = const [];
  void operationB(List<Item> show) {
    view = List<Item>.unmodifiable(show);
    m.notify(this, 'B:RESULTS_RENDERED', show.length);
  }
}
